### PR TITLE
OneDNN: report error for dilation max_pooling and replace AT_ERROR with TORCH_CHECK in oneDNN codes

### LINKED ISSUE #43514: MaxPool2d throwing RuntimeError for mkldnn tensor with certain parameters
## 🐛 Bug

Getting the following error when trying to call MaxPool2d with an mkldnn tensor created with .to_mkldnn() method.

__RuntimeError: could not create a descriptor for a pooling forward propagation primitive__

## To Reproduce

```
import torch

# Tensor Parameters
BATCH = 10
CHANNEL = 16
HEIGHT = 2048
WIDTH = 2048
DTYPE = torch.float32
DEVICE = "cpu"

# Pooling Parameters
KERNEL_SIZE = 4
STRIDE = 1
PADDING = 1
DILATION = 2
CEIL_MODE = True

blob = torch.randn(BATCH, CHANNEL, HEIGHT, WIDTH, dtype=DTYPE, device=DEVICE)

model = torch.nn.MaxPool2d(KERNEL_SIZE, STRIDE, PADDING, DILATION, False, CEIL_MODE)
model(blob.to_mkldnn())
```

## Expected behavior

Should compute max_pool2d correctly.

## Environment

Collecting environment information...
PyTorch version: 1.7.0a0+ccce112
Is debug build: False
CUDA used to build PyTorch: 10.1

OS: Ubuntu 18.04.3 LTS (x86_64)
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
Clang version: Could not collect
CMake version: version 3.14.0

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: 10.1.105
GPU models and configuration: 
GPU 0: Quadro GP100
GPU 1: Quadro GP100

Nvidia driver version: 418.116.00
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] torch==1.7.0a0+ccce112
[conda] blas                      1.0                         mkl  
[conda] magma-cuda101             2.5.2                         1    pytorch
[conda] mkl                       2020.1                      217  
[conda] mkl-include               2020.1                      217  
[conda] mkl-service               2.3.0            py38he904b0f_0  
[conda] mkl_fft                   1.1.0            py38h23d657b_0  
[conda] mkl_random                1.1.1            py38h0573a6f_0  
[conda] numpy                     1.18.5           py38ha1c710e_0  
[conda] numpy-base                1.18.5           py38hde5b4d6_0  
[conda] torch                     1.7.0a0+ccce112           dev_0    <develop>

## Additional context

I'm working on improving the pooling code so I'll look into this issue.

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @VitalyFedyunin

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Fix #43514.

### CODE REVIEW

File: aten/src/ATen/native/mkldnn/Pooling.cpp [hunk 1]
```diff
@@ -184,6 +184,10 @@ Tensor mkldnn_max_pool2d(
     IntArrayRef padding,
     IntArrayRef dilation,
     bool ceil_mode) {
+  TORCH_CHECK(dilation.size() == 1 || dilation.size() == 2,
+      "mkldnn_max_pool2d: dilation must be either be a single int, or a tuple of two ints");
+  TORCH_CHECK(dilation[0] == 1 && (dilation.size() == 2 ? dilation[0] == 1 : true),
```
Comment (ngimel) [inline aten/src/ATen/native/mkldnn/Pooling.cpp:189]: std::all_of, same below

**Review by ngimel — COMMENTED** [review]
Does mkl plan to support dilated pooling?

**Comment by XiaobingSuper** [comment]
> Does mkl plan to support dilated pooling?

Dilated pooling will be avilabled in OneDNN v2.0 release.

File: aten/src/ATen/native/mkldnn/Pooling.cpp (hunk 1 above)
Comment (XiaobingSuper) [reply aten/src/ATen/native/mkldnn/Pooling.cpp]: Changed.

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
Can you please add a test that catches the error that you are throwing for a dilated convolution case?

**Comment by XiaobingSuper** [comment]
> Can you please add a test that catches the error that you are throwing for a dilated convolution case?

Dilated max_pooling case is added.

File: aten/src/ATen/native/mkldnn/Pooling.cpp [hunk 2]
```diff
@@ -201,6 +203,8 @@ Tensor mkldnn_max_pool3d(
     IntArrayRef padding,
     IntArrayRef dilation,
     bool ceil_mode) {
+  TORCH_CHECK(std::all_of(dilation.cbegin(), dilation.cend(), [](int64_t i) { return 1 == i; }),
+      "mkldnn_max_pool3d not support dilation case");
```
Comment (heitorschueroff) [inline aten/src/ATen/native/mkldnn/Pooling.cpp:207]: typo: does not support

**Review by heitorschueroff — APPROVED** [review]

**Review by ngimel — APPROVED** [review]

File: test/test_mkldnn.py [hunk 3]
```diff
@@ -273,6 +273,33 @@ def test_max_pool3d(self):
...
+
+        # 2d dilation case
+        x = torch.randn(N, C, 7, 7, dtype=torch.float32).to_mkldnn()
+        max_pool2d = torch.nn.MaxPool2d(
+            kernel_size=3,
+            stride=3,
+            padding=1,
+            dilation=2)
+        self.assertRaisesRegex(RuntimeError,
+                               'mkldnn_max_pool2d dose not support dilation case',
```
Comment (ngimel) [inline test/test_mkldnn.py:289]: there's a typo here, so the tests are failing

Comment (XiaobingSuper) [reply test/test_mkldnn.py]: Sorry for this mistake. Changed.

**Pull request closed without merging**